### PR TITLE
 JAMES-2943 Deleting auto detected domain has no effect 

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[53b143a074ea6192c74cd6537c35a3beecd22499] 05:12:50.862 [ERROR] spark.Spark - ignite failed
[53b143a074ea6192c74cd6537c35a3beecd22499] java.net.BindException: Address already in use
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/sun.nio.ch.Net.bind0(Native Method)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/sun.nio.ch.Net.bind(Net.java:461)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/sun.nio.ch.Net.bind(Net.java:453)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:342)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	... 9 common frames omitted
[53b143a074ea6192c74cd6537c35a3beecd22499] Wrapped by: java.io.IOException: Failed to bind to /0.0.0.0:33885
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:346)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:308)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:236)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.server.Server.doStart(Server.java:396)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at spark.embeddedserver.jetty.EmbeddedJettyServer.ignite(EmbeddedJettyServer.java:149)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at spark.Service.lambda$init$2(Service.java:574)
[53b143a074ea6192c74cd6537c35a3beecd22499] 	at java.base/java.lang.Thread.run(Thread.java:834)
```

test this please

File: server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java [hunk 1]
```diff
@@ -176,6 +179,24 @@ public boolean containsDomain(Domain domain) throws DomainListException {
...
+            .addAll(domains)
+            .addAll(detectedDomains)
+            .build();
+        ImmutableList<Domain> ips = detectIps(domainsWithoutIp);
+
+        ImmutableMultimap.Builder<DomainType, Domain> result = ImmutableMultimap.<DomainType, Domain>builder()
+            .putAll(DomainType.Internal, domains)
+            .putAll(DomainType.Detected, detectedDomains)
+            .putAll(DomainType.DetectedIp, ips);
+        Optional.ofNullable(defaultDomain).ifPresent(domain -> result.put(DomainType.DefaultDomain, domain));
```
Comment (blackheaven) [inline server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java]: CR

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
CI restart...

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**